src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,715 @@
+    try:
+        non_slot_settings = []
+        slot_settings = []
+        for kv in key_values:
+            name = kv.key
+            value = kv.value
+            if 'AppService:SlotSetting' in kv.tags and kv.tags['AppService:SlotSetting'] == 'true':
+                slot_settings.append(name + '=' + value)
+            else:
+                non_slot_settings.append(name + '=' + value)
+        # known issue 4/26: with in-place update, AppService could change slot-setting true/false incorrectly
+        from azure.cli.command_modules.appservice.custom import update_app_settings
+        update_app_settings(cmd, resource_group_name=appservice_account["resource_group"],
+                            name=appservice_account["name"], settings=non_slot_settings, slot_settings=slot_settings)
+    except Exception as exception:
+        raise CLIError(
+            "Fail to write key-values to appservice: " + str(exception))
+
+
+# Helper functions
+
+
+def __serialize_kv_list_to_comparable_json_object(keyvalues, level):
+    res = {}
+    if level == 'file':  # import/export only key and value
+        for kv in keyvalues:
+            kv_json = {'value': kv.value}
+            res[kv.key] = kv_json
+    # import/export key, value, and tags (same level as key-value)
+    elif level == 'appservice':
+        for kv in keyvalues:
+            kv_json = {'value': kv.value}
+            if kv.tags:
+                for tag_k, tag_v in kv.tags.items():
+                    if tag_k == 'AppService:SlotSetting':
+                        kv_json[tag_k] = tag_v
+            res[kv.key] = kv_json
+    # import/export key, value, content-type, and tags (as a sub group)
+    elif level == 'appconfig':
+        for kv in keyvalues:
+            # value
+            kv_json = {'value': kv.value}
+            # tags
+            tag_json = {}
+            if kv.tags:
+                for tag_k, tag_v in kv.tags.items():
+                    tag_json[tag_k] = tag_v
+            kv_json['tags'] = tag_json
+            # content type
+            if kv.content_type:
+                kv_json['content type'] = kv.content_type
+            else:
+                kv_json['content type'] = ""
+            # key
+            res[kv.key] = kv_json
+    return res
+
+
+def __serialize_features_from_kv_list_to_comparable_json_object(keyvalues):
+    res = {}
+    for kv in keyvalues:
+        feature = map_keyvalue_to_featureflag(kv)
+        # state
+        feature_json = {'state': feature.state}
+        # conditions
+        feature_json['conditions'] = feature.conditions
+        # key
+        res[feature.key] = feature_json
+    return res
+
+
+def __serialize_feature_list_to_comparable_json_object(features):
+    res = {}
+    for feature in features:
+        # state
+        feature_json = {'state': feature.state}
+        # description
+        feature_json['description'] = feature.description
+        # conditions
```

Comment:
why we compare description in here but not in __serialize_features_from_kv_list_to_comparable_json_object

Reply:
Oh I didn't realize __serialize_features_from_kv_list_to_comparable_json_object is dead code now. Removing it. As for the description, we should still compare it in case features are being exported/imported between two config stores.

---
In reply to: [356898632](https://github.com/Azure/azure-cli/pull/11467#discussion_r356898632) [](ancestors = 356898632)